src/v2/coins/abstractUtxoCoin.js

```diff
@@ -517,41 +513,78 @@ class AbstractUtxoCoin extends BaseCoin {
   }
 
   /**
-   * Indicates whether a coin supports pay-to-witness script hash addresses
+   * Indicates whether a coin supports pay-to-witness-script-hash outputs wrapped in pay-to-script-hash outputs
+   * @returns {boolean}
+   */
+  supportsP2shP2wsh() {
+    return false;
+  }
+
+  /**
+   * Indicates whether a coin supports pay-to-witness-script-hash outputs
    * @returns {boolean}
    */
   supportsP2wsh() {
     return false;
   }
 
   /**
+   * TODO(BG-11487): Remove addressType, segwit, and bech32 params in SDKv6
    * Generate an address for a wallet based on a set of configurations
-   * @param addressType
-   * @param keychains Array of objects with xpubs
-   * @param threshold Minimum number of signatures
-   * @param chain Derivation chain
-   * @param index Derivation index
-   * @param segwit
-   * @param bech32
+   * @param addressType {string}   Deprecated
+   * @param keychains   {[object]} Array of objects with xpubs
+   * @param threshold   {number}   Minimum number of signatures
+   * @param chain       {number}   Derivation chain (see https://github.com/BitGo/unspents/blob/master/src/codes.ts for
+   *                                                 the corresponding address type of a given chain code)
+   * @param index       {number}   Derivation index
+   * @param segwit      {boolean}  Deprecated
+   * @param bech32      {boolean}  Deprecated
    * @returns {{chain: number, index: number, coin: number, coinSpecific: {outputScript, redeemScript}}}
    */
   generateAddress({ addressType, keychains, threshold, chain, index, segwit, bech32 }) {
-    if (addressType === this.constructor.AddressTypes.P2WSH && !this.supportsP2wsh()) {
-      throw new errors.P2wshUnsupportedError();
+    let derivationChain = 0;
+    if (_.isInteger(chain) && chain > 0) {
+      derivationChain = chain;
     }
 
-    if (!_.isUndefined(addressType) && !this.constructor.validAddressTypes.includes(addressType)) {
-      throw new errors.UnsupportedAddressTypeError();
-    } else if (_.isUndefined(addressType)) {
-      addressType = this.constructor.AddressTypes.P2SH;
+    if (_.isUndefined(addressType)) {
+      addressType = Codes.typeForCode(derivationChain);
       if (_.isBoolean(segwit) && segwit) {
-        addressType = this.constructor.AddressTypes.P2SH_P2WSH;
+        addressType = Codes.UnspentTypeTcomb('p2shP2wsh');
       }
       if (_.isBoolean(bech32) && bech32) {
```

Comment:
Since nobody ever actually used `bech32`, could we perhaps replace this with `p2wsh` from the get-go?

Reply:
I don't think that gets us anything. If we feel confident that this is not being used by customers then I would suggest removing it altogether as `chain` should be the canonical source. Otherwise we should just leave it as is.

Reply:
How would anyone be using `bech32` when it was always disabled to date?

Reply:
Clarified.